src/com/post/client/Sale.java

```diff
@@ -35,27 +35,19 @@ public Sale()
         
     }
     
-    public boolean addLineItem(ProductSpecification productSpecs, int quantity)
-    {
-        LineItem lineItem = new LineItem(productSpecs, quantity);
-        this.lineItems.add(lineItem);
-        return true;
-    }
-    
-    public boolean initiateSale(String custName, ArrayList<LineItem> lineItems)
+    public boolean initiateSale(String custName)
     {
        this.custName = custName;
-       this.lineItems = lineItems;
        this.date = new Date();
-       for(LineItem li : lineItems)
-       {
-           amountDue += li.getQuantity() * li.getItemUnitCost();
-       }
-       totalCost += amountDue;
-       
-       this.lineItems.addAll(lineItems);
        return true;
     }
+    
+    public LineItem addLineItem(ProductSpecification productSpecs, int quantity)
```

Comment:
I think it will would make more sense for the signature to look like this:

public LineItem addLineItem(LineItem li)

The Frame will create the line item and pass it to the Sale